[Heng Fu Yuen] iP

HengFuYuen opened the pull request.

HengFuYuen committed:
Add greetings and exits

HengFuYuen committed:
Revert "Add greetings and exits"

This reverts commit cf1ea773d454254c9d209a0192d1814bd534d6d8.

HengFuYuen committed:
Add greetings and exits

Duke is able to greet the user and exit after that

HengFuYuen committed:
Add echo feature

Duke starts by greeting the user before
echoing commands entered by the user
and exits when the user inputs bye.

HengFuYuen committed:
Add ability to store and display inputs

Duke can now store whatever text entered by the user and
display them back to the user when requested.

HengFuYuen committed:
Add the Task class and ability to mark tasks as done

HengFuYuen committed:
Follow Coding Standards

Code now complies with the coding standards.

HengFuYuen committed:
Add Deadline.java Event.java Todo.java

Supports tracking of three types of tasks, namely tasks with deadlines,
event tasks that start and end at specific timinings and ToDo tasks without dates attached.
Task.java and Duke.java are updated to accomodate the new classes.

HengFuYuen committed:
Automate Text UI Testing using I/O redirection

HengFuYuen committed:
Re-add the ability to store whatever text entered by the user

Level-2 feature of storing all user input was removed in previous commits
when other tasks types were added but this feature is now readded back.

HengFuYuen committed:
Add more test cases and compile all files

Additional test cases was added into input.txt and EXPECTED.TXT to test for
Level-2 feature of storing all inputs by users.
Runtest.bat now compiles all Java files rather than just Task.java.

HengFuYuen committed:
Improve code quality by refactoring

thatseant commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,137 @@
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        printWelcomeMessage();
+        while (true) {
+            String command = getUserCommand();
+            processUserCommand(command);
+        }
```
Good Job! 👍

I like how you abstracted your entire code into functions, allowing for each section to be readable and nameable without need for much comments.

thatseant reviewed (commented):
Good Job! Your code readability is good and follows coding guidelines!

kaijiel24 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,137 @@
+        System.out.println("____________________________________________________________");
+    }
+
+    private static Task addNewTask(String command) {
+        Task newTask;
+        if (command.startsWith("todo")) {
+            newTask = createTodo(command);
+        } else if (command.startsWith("deadline")) {
+            newTask = createDeadline(command);
+        } else if (command.startsWith("event")) {
+            newTask = createEvent(command);
+        } else {
+            newTask = createTask(command);
+        }
+        tasks[taskNumber] = newTask;
+        taskNumber++;
+        return newTask;
+    }
```
Well done :) I think it is good that you have abstracted the individual commands into their own function instead of putting them all in this function.

kaijiel24 reviewed (commented):
Well done :)

HengFuYuen committed:
Handle errors using Exceptions

Several custom Exception classes were added to
help handle unchecked errors that are specific to Duke.

HengFuYuen committed:
Merge branch 'branch-Level-5'

HengFuYuen committed:
Divide classes into packages

HengFuYuen committed:
Merge branch 'branch-A-Packages'

HengFuYuen committed:
Store tasks in ArrayList and add TaskList.java

TaskList.java handles all operations relating to the the task list.
For example it handles addition of new tasks to task list, marking tasks in the list as done, etc.
Duke.java only handles reading of inputs, resolving exceptions and printing of messages to user.

HengFuYuen committed:
Revert "Store tasks in ArrayList and add TaskList.java"

This reverts commit 0cc959d4f67694f18085d75b71b9522951aa3c08.

HengFuYuen committed:
Store tasks in ArrayList and add TaskList.java

TaskList.java handles all operations relating to the the task list.
For example it handles addition of new tasks to task list, marking tasks in the list as done, etc.
Duke.java only handles reading of inputs, resolving exceptions and printing of messages to user.

HengFuYuen committed:
Add support for deleting tasks from the list

HengFuYuen committed:
Add ability to save all tasks into the hard disk

Save the tasks in the hard disk automatically whenever the task list changes.
Load the data from the hard disk when Duke starts up.

HengFuYuen committed:
Add clearing tasklist ability and CommandWord.java

CommandWord.java is an enum class that stores command word strings as constants.

HengFuYuen committed:
Merge branch 'branch-Level-6'

HengFuYuen committed:
Improve code quality Duke.java TaskList.java

HengFuYuen committed:
Improve code quality of Duke.java and Task.java

manaswini05 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,334 @@
+    private static Task[] tasks = new Task[MAX_NUMBER_OF_TASKS];
+    private static int taskNumber = 0;
+
+    private static final String inputInstruction = "Please input in the following format:";
+    private static final String printTaskListInputFormat = "  To list all tasks: list";
```
constants have a different formatting style i.e. all caps and underscores. Consider changing into that

HengFuYuen replied:
Noted! Thanks for the advice!

manaswini05 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,334 @@
+    private static final String printTaskListInputFormat = "  To list all tasks: list";
+    private static String markTaskAsDoneInputFormat = "  To mark a task as done: done <task number>"
+            + System.lineSeparator() + "      Range of valid task number: ";
+    private static final String todoInputFormat = "  To add a todo: todo <description of todo>";
+    private static final String deadlineInputFormat = "  To add a deadline: deadline <description of "
+            + "deadline> /by <time it is due>";
+    private static final String eventInputFormat = "  To add an event: event <description of event> /at "
+            + "<the event time>";
+    private static final String exitDukeInputFormat = "  To exit duke.Duke: bye";
```
Do look into the constants naming format

manaswini05 reviewed (commented):
Great work! :) Consider adding comments in the recommended way

HengFuYuen committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/duke/Duke.java
#	text-ui-test/EXPECTED.TXT

HengFuYuen committed:
Refactor out Storage class

Storage.java deals with loading tasks from the file and saving tasks in the file.

HengFuYuen committed:
Remove wildcard import of Exception classes

HengFuYuen committed:
Handle too many timings given error

Deadline and Event tasks require a time input in addition to a task description.
However, as users may key in more than one timing for the task, this error must be handled.

HengFuYuen committed:
Refactor to extract Ui.java

Ui.java deals with interactions with the user.

HengFuYuen committed:
Refactor to extract Parser and Command classes

Parser.java deals with makiing sense of the user command.
Each valid user command has its own class.

HengFuYuen committed:
Remove unnecessary test cases

HengFuYuen committed:
Add ability to find tasks using a keyword

HengFuYuen committed:
Add JavaDoc comments to the code

HengFuYuen committed:
Add comments to code related to finding tasks

HengFuYuen committed:
Update JavaDoc comment for a method in Ui.java

HengFuYuen committed:
Merge pull request #1 from HengFuYuen/branch-Level-9

Merge branch "branch-Level-9"

HengFuYuen committed:
Merge branch 'master' of https://github.com/HengFuYuen/ip

HengFuYuen committed:
Merge branch 'master' into branch-A-JavaDoc

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/command/ByeCommand.java
#	src/main/java/duke/command/ClearCommand.java
#	src/main/java/duke/command/DeadlineCommand.java
#	src/main/java/duke/command/DeleteCommand.java
#	src/main/java/duke/command/DoneCommand.java
#	src/main/java/duke/command/EventCommand.java
#	src/main/java/duke/command/ListCommand.java
#	src/main/java/duke/command/TodoCommand.java
#	src/main/java/duke/task/Deadline.java
#	src/main/java/duke/task/Event.java
#	src/main/java/duke/task/Task.java

HengFuYuen committed:
Merge pull request #2 from HengFuYuen/branch-A-JavaDoc

Merge branch "branch-A-JavaDoc"

HengFuYuen committed:
Add a User Guide

HengFuYuen committed:
Set theme jekyll-theme-slate

HengFuYuen committed:
Set theme jekyll-theme-dinky

HengFuYuen committed:
Update _config.yml

HengFuYuen committed:
Make minor formatting adjustments to user guide

HengFuYuen committed:
Merge branch 'master' of https://github.com/HengFuYuen/ip

HengFuYuen committed:
Make minor formatting adjustments to user guide
